### PR TITLE
Python metadata plugin

### PR DESCRIPTION
**Pull request by vitsalis**
## Description
Introduce metadata plugin for inserting Python call graphs into the metadata database.
This pull request is based on #91 , and therefore it should be merged after it.

## Motivation and context
This is Python specific code which is very similar to the ones implemented for Java and C.

## Testing
1. Unit tests
2. Local tests on my machine (OS x) using a kafka topic which stored either the call graphs themselves or a path to the call graph.
3. Repeated the same tests on the monster server.

## Additional context
Based on #91 .

### CODE REVIEW

File: analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDBExtension.java [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+        // Create a GID Graph for production
+        this.gidGraph = new GidGraph(packageVersionId, callGraph.product, callGraph.version,
+                callablesIds, numInternal, edges);
+        return packageVersionId;
+    }
+
+    public ArrayList<CallablesRecord> insertDataExtractCallables(
+            ExtendedRevisionCallGraph callgraph, MetadataDao metadataDao, long packageVersionId) {
+        return new ArrayList<CallablesRecord>();
```
Comment (MihhailSokolov) [inline analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDBExtension.java:266]: I assume this is a TODO? :)

File: core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java [hunk 2]
```diff
@@ -330,6 +331,43 @@ public long insertModule(long packageVersionId, String namespace,
...
+     */
+    public long insertModule(long packageVersionId, String namespace,
+                             Timestamp createdAt, JSONObject metadata,
+                             boolean addDuplicates) {
+        var metadataJsonb = metadata != null ? JSONB.valueOf(metadata.toString()) : null;
+        var resultRecord = context.insertInto(Modules.MODULES,
+                Modules.MODULES.PACKAGE_VERSION_ID, Modules.MODULES.NAMESPACE,
+                Modules.MODULES.CREATED_AT, Modules.MODULES.METADATA)
+                .values(packageVersionId, namespace, createdAt, metadataJsonb)
+                // FIXME
```
Comment (MihhailSokolov) [inline core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java:352]: What is the problem here?

**Review by MihhailSokolov — CHANGES_REQUESTED** [review]
Overall looks good, but I do have some questions and a small comment

File: core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java (hunk 2 above)
Comment (vitsalis) [reply core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java:352]: This was added on #91 . @StefanosChaliasos could you please take a look?

File: analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDBExtension.java (hunk 1 above)
Comment (vitsalis) [reply analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDBExtension.java:266]: This was added on #91 . @StefanosChaliasos could you please take a look?

**Comment by vitsalis** [comment]
Thanks! Fixed the comment.

**Comment by vitsalis** [comment]
Closing this and creating a new pull request comparing with #91

File: analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDBExtension.java (hunk 1 above)
Comment (StefanosChaliasos) [reply analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDBExtension.java:266]: Nope, all classes that implement this class must provide an implementation for this method. Unfortunately, we cannot convert this class to an abstract class. I added a comment.

**Pull request closed without merging**